### PR TITLE
CONTRIBUTING, README: add contributing documentation

### LINKED ISSUE #78: Update README to be more user-focused
The `README.md` should include the output of `pip-audit --help` as well as provide some of the most common examples of how to use `pip-audit`.

The development guide should be moved to `CONTRIBUTING.md`.

### PR DESCRIPTION
**Pull request by woodruffw**
Remove from README, replace with installation steps.

Closes #78.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,38 @@ support from Google. This is not an official Google product.
 pip install pip-audit
```
Comment (di) [inline README.md:20]:
```suggestion
python -m pip install pip-audit
```

File: README.md [hunk 2]
```diff
@@ -4,45 +4,54 @@ pip-audit
 ![CI](https://github.com/trailofbits/pip-audit/workflows/CI/badge.svg)
 [![PyPI version](https://badge.fury.io/py/pip-audit.svg)](https://badge.fury.io/py/pip-audit)
 
-`pip-audit` is a prototype tool for scanning Python environments for packages with known vulnerabilities. It uses the Python Packaging Advisory Database (https://github.com/pypa/advisory-db) as a source of vulnerability reports.
+`pip-audit` is a prototype tool for scanning Python environments for packages
+with known vulnerabilities. It uses the Python Packaging Advisory Database
+(https://github.com/pypa/advisory-db) as a source of vulnerability reports.
```
Comment (di) [inline README.md:9]: Let's add something like "via the PyPI JSON API" and link to https://warehouse.pypa.io/api-reference/json.html.

File: README.md [hunk 3]
```diff
@@ -4,45 +4,54 @@ pip-audit
...
+  --desc {on,off,auto}  include a description for each vulnerability; `auto`
+                        only includes a description for the `json` format
+                        (default: auto)
+  --cache-dir CACHE_DIR
+                        the directory to use as an HTTP cache for PyPI; uses
+                        the `pip` HTTP cache by default (default: None)
 ```
 
-Run that last command sequence to complete the release.
-
```
Comment (di) [inline README.md:45]: Let's also add a "Contributing" section here and link to the new contributing docs.

**Review by di — COMMENTED** [review]
Can we also add some example runs w/ output, e.g. what it looks like to run `pip-audit` on an environment that has no vulnerabilities, what it looks like when it finds a vulnerability, running it on a requirements file, etc. -- just the most common ones.

File: README.md (hunk 2 above)
Comment (tetsuo-cpp) [reply README.md]: 78315f6

File: README.md (hunk 3 above)
Comment (tetsuo-cpp) [reply README.md:45]: 101a306

**Review by tetsuo-cpp — APPROVED** [review]
LGTM!

**Pull request merged**